File: codility/src/main/java/codility/com/lessons/l05prefixsums/PassingCars.java
package codility.com.lessons.l05prefixsums;

/**
 * 100%
 * @author 08115341622
 */
public class PassingCars {
	public int solution(int[] A) {
        long contUm[] = new long[A.length];
        contUm[A.length - 1] = A[A.length - 1] == 1 ? 1 : 0;
        for (int i = A.length - 2; i >= 0; i--) {
        	contUm[i] = contUm[i + 1] +  (A[i] == 1 ? 1 : 0);
        }
        long contador = 0;
        for (int i = 0; i < A.length; i++) {
        	if (A[i] == 0) {
        		if (1000000000 - contador < contUm[i]) {
        			contador = -1;
        			break;
        		}
        		contador += contUm[i];
        	}
        }
        return (int)contador;
    }
}

File: codility_new/passing_cars.py
# https://app.codility.com/programmers/lessons/5-prefix_sums/passing_cars/
def solution(A):
    ones_after_here = [0 for _ in range(len(A) + 1)]
    for x in A:
        if x == 1:
            ones_after_here[0] += 1
    
    for i in range(len(A)):
        if A[i] == 0:
            ones_after_here[i + 1] = ones_after_here[i]
        else:
            ones_after_here[i + 1] = ones_after_here[i] - 1
    resp = 0
    for i in range(len(A)):
        if A[i] == 0:
            resp += ones_after_here[i+1]
            if resp > 1_000_000_000:
                return -1
    return resp
